hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -481,16 +486,20 @@ public boolean isValidNode(DatanodeDetails datanodeDetails,
     if (datanodeInfo == null) {
       LOG.error("Failed to find the DatanodeInfo for datanode {}",
           datanodeDetails);
-    } else {
-      if (datanodeInfo.getNodeStatus().isNodeWritable() &&
-          (hasEnoughSpace(datanodeInfo, metadataSizeRequired,
-              dataSizeRequired))) {
-        LOG.debug("Datanode {} is chosen. Required metadata size is {} and " +
-                "required data size is {}",
-            datanodeDetails, metadataSizeRequired, dataSizeRequired);
-        return true;
-      }
+      return false;
+    }
+    NodeStatus nodeStatus = datanodeInfo.getNodeStatus();
+    if (nodeStatus.isNodeWritable() &&
+        (hasEnoughSpace(datanodeInfo, metadataSizeRequired,
+            dataSizeRequired))) {
+      LOG.debug("Datanode {} is chosen. Required metadata size is {} and " +
+              "required data size is {} and NodeStatus is {}",
+          datanodeDetails, metadataSizeRequired, dataSizeRequired, nodeStatus);
+      return true;
     }
+    LOG.debug("Datanode {} is not chosen. Required metadata size is {} and " +
```

Comment:
NIT: Do we need to wrap this one and line 495 above with a `LOG.isDebugEnabled()` check? `toString()` for `NodeStatus` does string concatenation using `+`. I suspect the compiler optimises this by using a `StringBuilder`, but not sure.

```
  public String toString() {
    return "OperationalState: " + operationalState + " Health: " + health +
        " OperationStateExpiry: " + opStateExpiryEpochSeconds;
  }
```
It's only using existing variables so I don't think this is an expensive computation anway tbh.

Reply:
Inside the LOG.debug method, it will only call the toString() method if the log is going to be produced. At parameter passing time, just the reference to the object is passed, and it should be passed as type Object, so no toString conversion gets done:

```  public void debug(String format, Object arg1, Object arg2) {
    if (this.logger.isDebugEnabled()) {
      FormattingTuple ft = MessageFormatter.format(format, arg1, arg2);
      this.logger.log(FQCN, Level.DEBUG, ft.getMessage(), ft.getThrowable());
    }

  }
```

The place you need to use `LOG.isDebugEnabled()` is when you need to build the string and pass that built string into the logger, eg:

```
String s = some_concatenations_or_logic_to_construct
LOG.debug("some log {}", s);
```

This is better written as:

```
if (LOG.isDebugEnabled()) {
  String s = some_concatenations_or_logic_to_construct
  LOG.debug("some log {}", s);
}
```